packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -253,8 +253,8 @@ class MDnsVmServiceDiscovery {
 
       _logger.printTrace(stackTrace.toString());
 
-      throwToolExit(
-        'Flutter could not connect to the Dart VM service.\n'
+      throw MissingLocalNetworkPermissionsException(
+        'Flutter could not access the local network.\n'
```

Comment:
I changed the error message because in some situations in can still connect to the Dart VM even if mDNS fails